hathor/cli/run_node.py

```diff
@@ -232,13 +246,71 @@ def register_signal_handlers(self) -> None:
         if sigusr1 is not None:
             # USR1 is available in this OS.
             signal.signal(sigusr1, self.signal_usr1_handler)
+        sigusr2 = getattr(signal, 'SIGUSR2', None)
+        if sigusr2 is not None:
+            # USR1 is available in this OS.
+            signal.signal(sigusr2, self.signal_usr2_handler)
 
     def signal_usr1_handler(self, sig: int, frame: Any) -> None:
         """Called when USR1 signal is received."""
         self.log.warn('USR1 received. Killing all connections...')
         if self.manager and self.manager.connections:
             self.manager.connections.disconnect_all_peers(force=True)
 
+    def signal_usr2_handler(self, sig: int, frame: Any) -> None:
```

Comment:
I think the implementation of this could be moved to a method in the runner, and make this method focus on not leaking any exceptions, with something like:

```python
    def signal_usr2_handler(self, sig: int, frame: Any) -> None:
        """Called when USR2 signal is received."""
        self.log.warn('USR2 received.')
        try:
            runner = self.get_sysctl_runner()
            runner.run_from_signal(self._args.data)
        except Exception:
            self.log.error('prevented exception from escaping the signal handler', exc_info=True)
```

Reply:
Nice suggestion. Done!